controller/setting_controller.go

```diff
@@ -211,15 +211,27 @@ func (sc *SettingController) handleErr(err error, key interface{}) {
 }
 
 func (sc *SettingController) syncSetting(key string) (err error) {
-	defer func() {
-		err = errors.Wrapf(err, "failed to sync setting for %v", key)
-	}()
-
 	_, name, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
 		return err
 	}
 
+	defer func() {
+		err = errors.Wrapf(err, "failed to sync setting for %v", key)
```

Comment:
This err should be still in the original defer closure, as it will be eventually executed.

Reply:
Moved back to the original defer closure.